### PR TITLE
fix to include the last byte in the uid2 in the hardware_uid string

### PR DESCRIPTION
**Pull request by DanielePettenuzzo**
I had this issue that when I update to the latest mavsdk the last character in my uid2 was not included in the hardware_uid in the info plugin. This is because snprintf doesn n-1 in the length argument. 

FYI @JonasVautherin

### CODE REVIEW

**Comment by DanielePettenuzzo** [comment]
My autopilot uuid: 000100000000313238313138511400360023
What I saw in the sdk before this pr: 00010000000031323831313851140036002

**Comment by DanielePettenuzzo** [comment]
I pushed a new commit. You mean something like this?

**Comment by julianoes** [comment]
> You mean something like this?

I think so, yes.

**Comment by DanielePettenuzzo** [comment]
I will test it again this morning

**Comment by DanielePettenuzzo** [comment]
I tested this with a few different pixhawk's and I always get he right uuid. I was wondering, is it enough to increase the size by 1 and std::string will add the null terminator or do I need to add the null terminator like in this pr?

**Comment by JonasVautherin** [comment]
> I was wondering, is it enough to increase the size by 1 and std::string will add the null terminator or do I need to add the null terminator like in this pr?

I would think that `std::string` doesn't require the `\0`. It will be added when you call `c_str()` or `data()`. But well, I'm the one who screwed up the `+ 1` here, so my feedback is to take with a grain of salt :see_no_evil:.

**Comment by julianoes** [comment]
This looks correct now in my testing.

File: src/plugins/info/info_impl.cpp [hunk 1]
```diff
@@ -172,12 +172,13 @@ std::string InfoImpl::swap_and_translate_binary_to_str(uint8_t* binary, unsigned
...
 std::string InfoImpl::translate_binary_to_str(uint8_t* binary, unsigned binary_len)
 {
-    std::string str(binary_len * 2, '0');
+    std::string str(binary_len * 2 + 1, '0');
 
     for (unsigned i = 0; i < binary_len; ++i) {
         // One hex number occupies 2 chars.
         snprintf(&str[i * 2], str.length() - i * 2, "%02x", binary[i]);
     }
+    snprintf(&str[binary_len * 2], str.length() - binary_len * 2 - 1, "%x", '\0');
```
Comment (JonasVautherin) [inline src/plugins/info/info_impl.cpp:181]:
```suggestion
```

I'm pretty sure it's superfluous to add `\0` at the end of an `std::string` (which is what we return here). As mentioned, `\0` is not part of an `std::string`, and it is just added when doing the `c_str()` conversion.

**Pull request merged**